### ISSUE TITLE
ebay-carousel: refactor status/headings API

### ISSUE THREAD

**Issue by ianmcburnie** [eBay/ebayui-core#1630]
Since the launch of carousel component, we have made a few learnings with regards to developer experience around a11y:

1. The "slide status" e.g. (Showing slide 1 of 3) would perhaps have been better placed on the pagination dots "group" than in the offscreen heading.
2. The live region around the slide status is perhaps unnecessary and too "noisy" 
3. The dots themselves (now removed) would be better as a separate component/module (e.g. ebay-carousel-pagination)
4. The offscreen heading would perhaps be better as not controlled by the component itself, and handled instead by the app. Any app provided heading can then be programmatically associated with the carousel via `aria-labelledby` (like how we do labelling of `ebay-istbox-button`)

Proposal (breaking changes):

1. Remove status completely
2. Create ebay-carousel-pagination (where status text would live)
3. Remove offscreen heading completely
4. Allow `aria-labelledby` pass through (to associate carousel with a heading element)

These changes I think would simplify use and expectations of carousel heading structure and a11y text.

**Comment by agliga** [comment]
For this we can have offscreen heading but allow aria-labelledby to be remapped. We can also have migrators to migrate the old a11yStatus and such text to point to offscreen heading, and then next major version we remove it.

**Comment by ianmcburnie** [comment]
Yeah, it's quite possible after we simplify a few things by removing status and live-region that we can keep the offscreen heading around (like we have for pagination and breadcrumbs).
